source/nodejs/adaptivecards-designer/src/app.ts

```diff
@@ -237,65 +264,56 @@ class DesignerApp {
         if (this.paletteHostElement) {
             this.paletteHostElement.innerHTML = "";
 
-            var sortedRegisteredTypes: Array<Adaptive.ITypeRegistration<Adaptive.CardElement>> = [];
-
-            for (var i = 0; i < Adaptive.AdaptiveCard.elementTypeRegistry.getItemCount(); i++) {
-                sortedRegisteredTypes.push(Adaptive.AdaptiveCard.elementTypeRegistry.getItemAt(i));
+            let sortedRegisteredTypes = {};
+
+            const categoriesMap = {
```

Comment:
Category should be a property of ITypeRegistration, you shouldn't maintain a separate map. Along the same lines, and to follow up on my comments about icons, ITypeRegistration should also have an iconUrl property.

Reply:
P0: category needs to be passed in constructor for peer registration - additional map as created here needs go